### ISSUE TITLE
Randomly Seeing *time.Time, not time.Time Error on Server

### ISSUE THREAD

**Issue by willcj33** [dgraph-io/dgraph#778]
```
interface conversion: interface is *time.Time, not time.Time
```
The above error happens on the server running in kubernetes causing a backoff restart cycle when the error is hit.

**Comment by pawanrawal** [comment]
Hi @willcj33 

Is this triggered by a specific mutation or query? Could you start dgraph with debugmode
`./dgraph -debugmode=true` and give us the stack trace assuming this is a panic.

**Comment by willcj33** [comment]
```
2017/03/29 20:02:51 lists.go:228: Dirty map size: 0
panic: interface conversion: interface is *time.Time, not time.Time

goroutine 108309 [running]:
panic(0xe34a40, 0xc422e78000)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/dgraph-io/dgraph/query.toProtoValue(0x5, 0xecaaa0, 0xc4278aa700, 0x0)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/proto.go:46 +0x495
github.com/dgraph-io/dgraph/query.createProperty(0xc4278aa6a0, 0x16, 0x5, 0xecaaa0, 0xc4278aa700, 0xecaa01)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/query.go:408 +0x41
github.com/dgraph-io/dgraph/query.(*protoNode).AddValue(0xc4227371c8, 0xc4278aa6a0, 0x16, 0xc400000005, 0xecaaa0, 0xc4278aa700)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/outputnode.go:90 +0x61
github.com/dgraph-io/dgraph/query.(*SubGraph).preTraverse(0xc42789b980, 0x8a23f92cade6fc65, 0x14ec5e0, 0xc4227371a8, 0x14ec5e0, 0xc422737118, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/query.go:262 +0x543
github.com/dgraph-io/dgraph/query.(*SubGraph).preTraverse(0xc42789ad80, 0xc2ee0e7383c36fc, 0x14ec5e0, 0xc422737118, 0x14ec5e0, 0xc422737118, 0xc424d5d858, 0x0)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/query.go:285 +0x9c4
github.com/dgraph-io/dgraph/query.(*SubGraph).ToProtocolBuffer(0xc42789ad80, 0xc424d5da38, 0x0, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/outputnode.go:195 +0x2ac
github.com/dgraph-io/dgraph/query.ToProtocolBuf(0xc424d5da38, 0xc4278a19a0, 0x3, 0x4, 0x4, 0xc4278a1720, 0x3, 0x3, 0x0)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/outputnode.go:46 +0xad
main.(*grpcServer).Run(0x15468a0, 0x7fb89c0f3700, 0xc42211b710, 0xc4278a13a0, 0xc42211b830, 0x0, 0x0)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/cmd/dgraph/main.go:599 +0x534
github.com/dgraph-io/dgraph/query/graph._Dgraph_Run_Handler(0xe22ce0, 0x15468a0, 0x7fb89c0f3700, 0xc42211b710, 0xc4278a2720, 0x0, 0x0, 0x0, 0x0, 0x11)
	/home/ubuntu/go/src/github.com/dgraph-io/dgraph/query/graph/graphresponse.pb.go:692 +0x27d
google.golang.org/grpc.(*Server).processUnaryRPC(0xc42017c480, 0x14ec9a0, 0xc42012add0, 0xc426fbca00, 0xc420011b60, 0x14d3cf0, 0xc42211b7d0, 0x0, 0x0)
	/home/ubuntu/go/src/google.golang.org/grpc/server.go:709 +0xbd8
google.golang.org/grpc.(*Server).handleStream(0xc42017c480, 0x14ec9a0, 0xc42012add0, 0xc426fbca00, 0xc42211b7d0)
	/home/ubuntu/go/src/google.golang.org/grpc/server.go:893 +0x6a4
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc42011fc30, 0xc42017c480, 0x14ec9a0, 0xc42012add0, 0xc426fbca00)
	/home/ubuntu/go/src/google.golang.org/grpc/server.go:469 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/ubuntu/go/src/google.golang.org/grpc/server.go:470 +0xa3
```
Seems to be after many many mutations.

**Comment by pawanrawal** [comment]
Not able to reproduce this. A nil value isn't allowed in a mutation so ideally this shouldn't happen. Could you please share some steps to reproduce this with data? Also, could you try doing the same query through the HTTP interface and see if it still happens?

**Comment by pawanrawal** [comment]
Hey @willcj33 

Could you try with v0.7.4 (released today) and see if you still get it? We want to resolve it with priority.

Thanks

**Comment by willcj33** [comment]
Yeah, I will take a look at that when I get in today.

**Comment by pawanrawal** [comment]
Was able to reproduce this on v0.7.3. If there was a scalar type which did not have any data and we did an aggregation query on it then we got this error. On v0.7.3, this test reproduces it, because although `dob1` is a scalar type, there are no values for it.

```
func TestMin1(t *testing.T) {
	populateGraph(t)
	// Alright. Now we have everything set up. Let's create the query.
	query := `
                {
                        me(id:0x01) {
                                name
                                gender
                                alive
                                friend {
                                   min(dob1)
                                }
                        }
                }
        `
	js := processToFastJSON(t, query)
}
```

Confirmed that this has been fixed on v0.7.4 hence closing this.
